### ISSUE TITLE
RabbitMQ - Add support for TLS connections (amqp+ssl) in connection factory

### ISSUE THREAD

**Issue by stephaneplante** [SteeltoeOSS/steeltoe#115]
Under RabbitMQ Tile  1.15 and higher, it is possible to enable TLS by specifying the proper configuration `cf update-service my-instance -c '{"tls": ["hostname_or_ip"]}'`.  Once this is done the Steeltoe Connector for RabbitMQ is not able to establish communication with the service instance.

The connections informations for a service with TLS enabled is not in the protocols:amqp section, it is now in the protocols:amqp+ssl.

Is it possible to add the support for this configuration?

Here is a sample of th service key credentials for a RabbitMQ instance with TLS enabled : 
```json
{
  "dashboard_url": "https://...",
  "hostname": "...",
  "hostnames": [
    "..."
  ],
  "http_api_uri": "https://...",
  "http_api_uris": ["..."],
  "password": "...",
  "protocols": {
    "amqp+ssl": {
      "host": "...",
      "hosts": ["..."],
      "password": "...",
      "port": 5671,
      "ssl": true,
      "uri": "amqps://...",
      "uris": ["amqps://..."],
      "username": "...",
      "vhost": "..."
    }
  },
  "ssl": true,
  "uri": "amqps://...",
  "uris": ["amqps://..."],
  "username": "...",
  "vhost": "..."
}


```

**Comment by TimHess** [comment]
OK, I finally got ssl setup in my environment, and the Steeltoe sample worked fine there, so something else might be going on in the environment - Could you try deploying [this Steeltoe sample](https://github.com/SteeltoeOSS/Samples/tree/master/Connectors/src/AspDotNetCore/RabbitMQ) and RabbitMQ with ssl bindings to see if that works for you? 

The platform has settings for this, if the sample doesn't work, I think the issues might be a bit outside of Steeltoe. For reference, I used [these instructions](https://docs.pivotal.io/rabbitmq-cf/1-16/prepare-tls.html) for getting my environment prepped, and my service binding came out like this:

```
"p-rabbitmq": [
        {
          "label": "p-rabbitmq",
          "provider": null,
          "plan": "standard",
          "name": "myRabbitMQService",
          "tags": [ "rabbitmq", "rabbit", "messaging", "message-queue", "amqp", "mqtt", "stomp" ],
          "instance_name": "myRabbitMQService",
          "binding_name": null,
          "credentials": {
            "dashboard_url": "https://pivotal-rabbitmq.<cfdomain>/#/login/user/pass",
            "hostname": "10.194.45.197",
            "hostnames": ["10.194.45.197" ],
            "http_api_uri": "https://user:pass@pivotal-rabbitmq.<cfdomain>/api/",
            "http_api_uris": ["https://user:pass@pivotal-rabbitmq.<cfdomain>/api/" ],
            "password": "pass",
            "protocols": {
              "amqp+ssl": {
                "host": "10.194.45.197",
                "hosts": ["10.194.45.197" ],
                "password": "pass",
                "port": 5671,
                "ssl": true,
                "uri": "amqps://user:pass@10.194.45.197:5671/ecfaa86e-ad76-405c-9d64-31f6782f5571",
                "uris": ["amqps://user:pass@10.194.45.197:5671/ecfaa86e-ad76-405c-9d64-31f6782f5571" ],
                "username": "user",
                "vhost": "ecfaa86e-ad76-405c-9d64-31f6782f5571"
              },
              "management": {
                "host": "10.194.45.197",
                "hosts": ["10.194.45.197"],
                "password": "pass",
                "path": "/api/",
                "port": 15672,
                "ssl": false,
                "uri": "http://user:pass@10.194.45.197:15672/api/",
                "uris": ["http://user:pass@10.194.45.197:15672/api/"],
                "username": "user"
              },
              "management+ssl": {
                "host": "10.194.45.197",
                "hosts": ["10.194.45.197"],
                "password": "pass",
                "path": "/api/",
                "port": 15672,
                "ssl": false,
                "uri": "http://user:pass@10.194.45.197:15672/api/",
                "uris": ["http://user:pass@10.194.45.197:15672/api/"],
                "username": "user"
              }
            },
            "ssl": true,
            "uri": "amqps://user:pass@10.194.45.197/ecfaa86e-ad76-405c-9d64-31f6782f5571",
            "uris": ["amqps://user:pass@10.194.45.197/ecfaa86e-ad76-405c-9d64-31f6782f5571"],
            "username": "user",
            "vhost": "ecfaa86e-ad76-405c-9d64-31f6782f5571"
          },
          "syslog_drain_url": null,
          "volume_mounts": []
        }
      ]
```

**Comment by stephaneplante** [comment]
Hi @TimHess,

Thank you very much for you help and time.  I had misread the code of the connector.  This request can be closed since the connectors already work perfectly well.

I finally found what the problem was.  In our infrastructure, there was a missing ASG regarding RabbitMQ.  The ports for TLS were closed so my app could not communicate with the service instance in TLS. As soon as the ports have been opened the rest worked out perfectly.

Again, Thanks!
